app/components/post_list/post_list.js

```diff
@@ -96,25 +99,64 @@ export default class PostList extends PureComponent {
         mattermostManaged.removeEventListener(this.listenerId);
     }
 
+    handleClosePermalink = () => {
+        const {actions} = this.props;
+        actions.selectFocusedPostId('');
+        this.setState({showingPermalink: false});
+    };
+
+    handlePermalinkPress = (postId) => {
+        this.setState({isPermalink: true});
```

Comment:
Is `this.state.isPermalink` needed? It looks like we set it here and then immediately reference it below which may not have the value you've set here if I understand React's state updating properly. If it does have the new value of `this.state.isPermalink` below, then we're always passing `isPermalink: true`  into the Permalink screen from here, so storing it in state is unnecessary.

If we do end up needing it in state, we should define an initial value since `this.state.isPermalink` is initially undefined